crates/cdk/src/nuts/nut02.rs

```diff
@@ -130,81 +131,46 @@ impl fmt::Display for Id {
     }
 }
 
-impl FromStr for Id {
-    type Err = Error;
+impl TryFrom<String> for Id {
+    type Error = Error;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        // Check if the string length is valid
+    fn try_from(s: String) -> Result<Self, Self::Error> {
```

Comment:
Is there any benefit to having a `TryFrom<String>` should we just remove it in favor of the FromStr`?

Reply:
`utoipa` needed conversion directly to/from `String`, otherwise it would show the `Id` as a struct with inner fields, not as a `String`.

I did find a way around that by manually deriving `utoipa::ToSchema`.

However it's rather verbose: https://github.com/cashubtc/cdk/compare/ed76e34707627ff076f6bf12f67f2f0f4881469f..2cab443cdc75d3833b27f66eefd78a65da356bbb

IMO the previous version was better. `TryFrom<String>` was just a wrapper around `FromStr`, so no duplication. Same for `impl From<Id> for String`.

What do you think, should I revert this last change?

Reply:
Makes sense if you're happier with the first version lets go with that.

Reply:
Done: https://github.com/cashubtc/cdk/compare/2cab443cdc75d3833b27f66eefd78a65da356bbb..2a092669264768cc2d58d3787a6b99658b10693f

Reply:
Thank you